interceptor/intercept.go

```diff
@@ -85,35 +91,108 @@ func (i *Interceptor) Intercept(nextHop string, reqPaymentHash []byte, reqOutgoi
 		}
 		log.Printf("paymentHash:%x\npaymentSecret:%x\ndestination:%x\nincomingAmountMsat:%v\noutgoingAmountMsat:%v",
 			paymentHash, paymentSecret, destination, incomingAmountMsat, outgoingAmountMsat)
-		if paymentSecret == nil || (nextHop != "<unknown>" && nextHop != hex.EncodeToString(destination)) {
+
+		isRegistered := paymentSecret != nil
+		isProbe := isRegistered && !bytes.Equal(paymentHash, reqPaymentHash)
+		nextHop, _ := i.client.GetPeerId(scid)
+		// If the payment was registered, but the next hop is not the destination
+		// that means we are not the last hop of the payment, so we'll just forward.
+		if destination != nil && nextHop != nil && !bytes.Equal(nextHop, destination) {
 			return InterceptResult{
 				Action: INTERCEPT_RESUME,
 			}, nil
 		}
 
-		if channelPoint == nil {
-			if bytes.Equal(paymentHash, reqPaymentHash) {
-				if int64(reqIncomingExpiry)-int64(reqOutgoingExpiry) < int64(i.config.TimeLockDelta) {
+		if nextHop == nil {
+			nextHop = destination
```

Comment:
Can you explain this? Not sure I fully understood.

Reply:
I should comment on that in the code, I will.
`destination` is the pubkey of the node that registered a payment, as extracted from the database. It's nil if there is no payment registered for this payment hash.
`nextHop` is the pubkey of the next node, as extracted from the onion. It's nil if the next hop is unknown in a 'normal' lightning flow.

At this point in the code it could still be a normal forward or a channel open. In both cases we want to notify if the peer is offline and registered for notications. 

If it's a channel open, that would mean the nexthop is nil and the destination is set. And effectively, the destination is the next hop. After this assignment, nextHop is the pubkey of the next hop, so we know who to notify. Or it's nil if the next hop is truly unknown.